src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -830,6 +831,32 @@ GraphPatternOperation Visitor::visitPathQuery(
   return pathQuery;
 }
 
+GraphPatternOperation Visitor::visitSpatialQuery(
+    Parser::ServiceGraphPatternContext* ctx) {
+  auto parseSpatialQuery = [](parsedQuery::SpatialQuery& spatialQuery,
+                              const parsedQuery::GraphPatternOperation& op) {
+    if (std::holds_alternative<parsedQuery::BasicGraphPattern>(op)) {
+      spatialQuery.addBasicPattern(
+          std::get<parsedQuery::BasicGraphPattern>(op));
+    } else if (std::holds_alternative<parsedQuery::GroupGraphPattern>(op)) {
+      spatialQuery.addGraph(op);
+    } else {
+      throw parsedQuery::SpatialSearchException(
+          "Unsupported element in spatialQuery."
+          "spatialQuery may only consist of triples for configuration"
+          "And a { group graph pattern } specifying the right join table.");
+    }
```

Comment:
you should use the `reportError` function, it automatically colors the wrong SPARQL construct in the UI.